### DISCUSSION TITLE
Ability to filter on nested read of nullable to-one relation field

### DISCUSSION THREAD

**Discussion by ymc9** [prisma/prisma#20350] [Q&A]
Hi guys,

I just noticed that with later versions of Prisma Client, I could put a where filter in the "include" clause when including a to-one relation if the field is optional. E.g.:

```prisma
model User {
  id      Int      @id @default(autoincrement())
  email   String   @unique
  name    String?
  profile Profile?
}

model Profile {
  id     Int     @id @default(autoincrement())
  bio    String?
  user   User    @relation(fields: [userId], references: [id], onDelete: Cascade)
  userId Int     @unique
}
```

At least starting from V4.8.0, after turning on the "extendedWhereUnique" preview flag, I can write a query like:
```ts
prisma.user.findUnique({
    where: { ... },
    include: {
        profile: { where: { ... } },  // <- filter here
    },
});
```

I wasn't able to find any information in the release notes or documentation. Just wondering if this is something that can be reliably used and will continue to be supported in the future version.

Thank you!

**Comment by janpio** [comment]
Yes, the preview feature was removed in 5.0.0 and the feature is now generally available: 
https://github.com/prisma/prisma/releases/tag/5.0.0#:~:text=Support%20for%20filtering%20non%2Dunique%20columns%20in%20queries%20for%20a%20unique%20record
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput

**Reply by ymc9** [reply]
Thanks @janpio . I knew that the "extendedWhereUnique" switch entered GA in V5, which is great. Just wasn't ware that this implied I could filter nullable to-one relations as shown in my example above. Maybe it was a side effect, but very nice to have. Just wanted to confirm it's part of the plan but not an accident (that'll potentially be reverted in the future) 😄.

**Reply by janpio** [reply]
Oh, ok - I did not actually pick up that subtle detail. I'll have to check about that.

What is your use case behind that? Find a specific user, but then only add the Profile if it has specific values?

**Reply by ymc9** [reply]
Thanks!

My use case is to conditionally include a nested to-one relation. There're access rules related to the "Profile" model that I wish to inject into the query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
